### ISSUE TITLE
Doesn't work in browser

### ISSUE THREAD

**Issue by Gbr22** [wasmerio/wasmer-js#293]
Here is my code:
```js
import { Buffer } from 'buffer'
window.Buffer = Buffer;

import { init, WASI } from '@wasmer/wasi';

await init();

let wasi = new WASI({
  env: {
      // 'ENVVAR1': '1',
      // 'ENVVAR2': '2'
  },
  args: [
    'command', 'arg1', 'arg2'
  ],
});

async function wasiTest(){
    const moduleBytes = fetch("/cowsay.wasm");
    const module = await WebAssembly.compileStreaming(moduleBytes);
    // Instantiate the WASI module
    await wasi.instantiate(module, {});

    // Run the start function
    let exitCode = wasi.start();
    let stdout = wasi.getStdoutString();

    // This should print "hello world (exit code: 0)"
    console.log(`${stdout}(exit code: ${exitCode})`);
}
```
When I call wasiTest I get the following error:
![image](https://user-images.githubusercontent.com/25722827/170833990-2fcd4521-8fdd-4291-af1c-b5f25a31b98b.png)
```
Uncaught (in promise) Error: Failed to instantiate WASI: RuntimeError: `
    at B2.wbg.__wbg_new_342a24ca698edd87 (Library.esm.min.js:25:11051)
    at 0024ee5a:0x7c9d3
    at 0024ee5a:0x19d5a
    at s.instantiate (Library.esm.min.js:25:6746)
    at wasiTest (GlobalState.js?t=1653754911321:26:16)
```

**Comment by Gbr22** [comment]
Update: It seems to work in firefox

**Comment by Gbr22** [comment]
TLDR: Move your code to a worker thread

I figured it out

I found a similar issue to mine: https://github.com/wasmerio/wasmer-js/issues/286#issuecomment-1099402450

Which led me to this issue: https://github.com/wasmerio/wasmer/issues/2792#issuecomment-1045986441

I downloaded and built [HarukaMa](https://github.com/HarukaMa)'s version and it threw this error message:
![image](https://user-images.githubusercontent.com/25722827/170835982-1f581bc5-6dd0-4728-81ee-b9d1051afe23.png)

Then I moved my code into a worker and it magically worked. Turns out there's no issue with the library after all it's just that it fails to give a helpful error message that I have to use workers

**Comment by corwin-of-amber** [comment]
I would comment that the loss of the original error string *is* a bug. It even looks like something was supposed to be there: there is an extra tick after `RuntimeError:`. WDYT, can someone look at the code throwing the error perhaps? Might be a simple typo.

**Comment by X7md** [comment]
> TLDR: Move your code to a worker thread
> 
> I figured it out
> 
> I found a similar issue to mine: [#286 (comment)](https://github.com/wasmerio/wasmer-js/issues/286#issuecomment-1099402450)
> 
> Which led me to this issue: [wasmerio/wasmer#2792 (comment)](https://github.com/wasmerio/wasmer/issues/2792#issuecomment-1045986441)
> 
> I downloaded and built [HarukaMa](https://github.com/HarukaMa)'s version and it threw this error message: ![image](https://user-images.githubusercontent.com/25722827/170835982-1f581bc5-6dd0-4728-81ee-b9d1051afe23.png)
> 
> Then I moved my code into a worker and it magically worked. Turns out there's no issue with the library after all it's just that it fails to give a helpful error message that I have to use workers

Could you put any example or code snippet, about how you actually move it to `Worker`?
